brave/browser/brave_browser_context.cc

```diff
@@ -162,6 +172,21 @@ BraveBrowserContext::BraveBrowserContext(
         atom::AtomBrowserContext::From(parent_partition, false));
   }
 
+  bool isolated_storage;
+  if (options.GetBoolean("isolated_storage", &isolated_storage)) {
+    isolated_storage_ = isolated_storage;
+  }
+
+  std::string tor_proxy;
+  if (options.GetString("tor_proxy", &tor_proxy)) {
+    tor_proxy_ = GURL(tor_proxy);
```

Comment:
The GURL parser doesn't handle the `socks5://...` notation sensibly, so for, say, `socks5://127.0.0.1:9050`, the object `tor_proxy_` ends up as a GURL with components

- scheme: `socks5`
- username: missing
- password: missing
- host: missing
- port: missing
- path: `//127.0.0.1:9050`

Later on, in proxy_config_service_tor.cc, we unparse this using `tor_proxy_.spec()`, which happens to reassemble it back into the string `socks5://127.0.0.1:9050` and then pass it on to the proxy config parser which does parse that notation as we expect.

Can we either skip parsing this URL here altogether, or use `url::ParseStandardURL` from url/third_party/mozilla/url_parse.h as the proxy config parser uses?  (For launching tor with a prescribed port number, it would be convenient to have the port number available here.)